qiskit/pulse/instructions/acquire.py

```diff
@@ -0,0 +1,240 @@
+                      result for fast feedback.
+            kernel: A ``Kernel`` for integrating raw data.
+            discriminator: A ``Discriminator`` for discriminating kerneled IQ data into 0/1
+                           results.
+            name: Name of the instruction for display purposes.
+
+        Raises:
+            PulseError: If channels are supplied, and the number of register and/or memory slots
+                        does not equal the number of channels.
+        """
+        if isinstance(channel, list) or isinstance(mem_slot, list) or reg_slots or mem_slots:
+            warnings.warn('The AcquireInstruction on multiple qubits, multiple '
+                          'memory slots and multiple reg slots is deprecated. The '
+                          'parameter "mem_slots" has been replaced by "mem_slot" and '
+                          '"reg_slots" has been replaced by "reg_slot"', DeprecationWarning, 3)
+
+        if not isinstance(channel, list):
+            channels = [channel] if channel else None
+        else:
+            channels = channel
+
+        if mem_slot and not isinstance(mem_slot, list):
+            mem_slot = [mem_slot]
+        elif mem_slots:
+            mem_slot = mem_slots
+
+        if reg_slot:
+            reg_slot = [reg_slot]
+        elif reg_slots and not isinstance(reg_slots, list):
+            reg_slot = [reg_slots]
+        else:
+            reg_slot = reg_slots
+
+        if channels and not (mem_slot or reg_slot):
+            raise PulseError('Neither MemorySlots nor RegisterSlots were supplied.')
+
+        if channels and mem_slot and len(channels) != len(mem_slot):
+            raise PulseError("The number of mem_slots must be equal to the number of channels.")
+
+        if channels and reg_slot:
+            if len(channels) != len(reg_slot):
+                raise PulseError("The number of reg_slots must be equal to the number of "
+                                 "channels.")
+        else:
+            reg_slot = []
+
+        if name is None and channels is None:
+            name = 'acq{:10x}'.format(hash((duration, kernel, discriminator)))
+        elif name is None:
+            name = 'acq{:10x}'.format(hash((duration, tuple(channels), tuple(mem_slot),
+                                            tuple(reg_slot), kernel, discriminator)))
+
+        if channels is not None:
+            super().__init__(duration, *channels, *mem_slot, *reg_slot, name=name)
+        else:
+            super().__init__(duration, name=name)
+
+        self._acquires = channels
+        self._channel = channels[0] if channels else None
+        self._mem_slots = mem_slot
+        self._reg_slots = reg_slot
+        self._kernel = kernel
+        self._discriminator = discriminator
+
+    @property
+    def operands(self) -> List:
+        """Return a list of instruction operands."""
+        return [self.duration, self.channel,
```

Comment:
I wonder if we should treat the `kernel` and `discriminator` as part of the instruction operands, or instruction metadata?

Reply:
oh, good point. I would say metadata, especially given the idea of moving it out of acquire